cmd/minishift/cmd/start.go

```diff
@@ -332,6 +335,22 @@ func startHost(libMachineClient *libmachine.Client) *host.Host {
 		fmt.Println("   Disk size:", units.HumanSize(float64(machineConfig.DiskSize*units.MB)))
 	}
 
+	// Experimental features
+	if minishiftConfig.EnableExperimental {
+		networkSettings := minishiftNetwork.NetworkSettings{
+			Device:    viper.GetString(configCmd.NetworkDevice.Name),
+			IPAddress: viper.GetString(configCmd.IPAddress.Name),
+			Netmask:   viper.GetString(configCmd.Netmask.Name),
+			Gateway:   viper.GetString(configCmd.Gateway.Name),
+			DNS1:      viper.GetString(configCmd.Nameserver.Name),
+		}
+
+		// Configure networking on startup only works on Hyper-V
+		if machineConfig.VMDriver == "hyperv" && networkSettings.IPAddress != "" {
+			minishiftNetwork.ConfigureNetworking(constants.MachineName, networkSettings)
+		}
+	}
```

Comment:
@gbraad We should check for CentOS or RHEL and communicate to user about if the options will not for users ISO else if user uses B2D it will fail and user wont have any clue unless user has read the documentation.

Reply:
I think there should be a validation function or something which would make sure to print right message for users if it is not a CentOS or RHEL ISO and on Windows HyperV

Reply:
windows only already... and here unsured to be hyperv only...

RHEL will support this...

Reply:
we can not check IsCDK/IsCentOS image, as this is done AFTER the the vm already started and starts provisioning... this is too late.

therefore the message is now: "Attempting to set network settings"... and the preflight check after host start can check for the final IP... but this is an improvement task. Remember this is experimental.